File: backend/models/comment.js
import mongoose from "mongoose";

const Schema = mongoose.Schema;

const commentSchema = new Schema(
  {
    commentType: Boolean, //true = summative, false = formative
    studentId: { type: Schema.Types.ObjectId, ref: "User" },
    postedBy: { type: Schema.Types.ObjectId, ref: "User" },
    body: String,
    datePosted: Date,
  }
);

const Comment = mongoose.model("Comment", commentSchema);

export default Comment;

File: gamification/Gamification/Assets/Scripts/Entities/Comment.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Comment
{
    public string Date { get; private set; }
    public string Body { get; private set; }

    public Comment(string date, string body)
    {
        Date = date;
        Body = body;
    }
}
